parallel_process_data.py

```diff
@@ -0,0 +1,600 @@
+                res['ceilings']['wind_speed']['rank'][25][i_out, i_lat, i_lon] = v_ranks[3]
+
+                p_mean, p_perc5, p_perc32, p_perc50 = get_statistics(p_ceiling)
+                res['ceilings']['wind_power_density']['mean'][i_out, i_lat, i_lon] = p_mean
+                res['ceilings']['wind_power_density']['percentile'][5][i_out, i_lat, i_lon] = p_perc5
+                res['ceilings']['wind_power_density']['percentile'][32][i_out, i_lat, i_lon] = p_perc32
+                res['ceilings']['wind_power_density']['percentile'][50][i_out, i_lat, i_lon] = p_perc50
+
+                p_ranks = get_percentile_ranks(p_ceiling, [40., 300., 1600., 9000.])
+                res['ceilings']['wind_power_density']['rank'][40][i_out, i_lat, i_lon] = p_ranks[0]
+                res['ceilings']['wind_power_density']['rank'][300][i_out, i_lat, i_lon] = p_ranks[1]
+                res['ceilings']['wind_power_density']['rank'][1600][i_out, i_lat, i_lon] = p_ranks[2]
+                res['ceilings']['wind_power_density']['rank'][9000][i_out, i_lat, i_lon] = p_ranks[3]
+
+    print('Locations analyzed: ({}/{}).'.format(counter, total_iters))
+    time_lapsed = float(timer()-start_time)
+    time_remaining = time_lapsed/counter*(total_iters-counter)
+    print("Time lapsed: {:.2f} hrs, expected time remaining: {:.2f} hrs.".format(time_lapsed/3600,
+                                                                                     time_remaining/3600))
+    return(res, counter)
+
+
+def process_complete_grid(output_file, subset_number_input):
+    """"Execute analyses on the data of the complete grid and save the processed data to a netCDF4 file.
+
+    Args:
+        output_file (str): Name of netCDF file to which the results are saved.
+        subset_number_input (int): Command line argument input choosing a specific latitude subset - (-1) for all subsets
+
+    """
+    ds, lons, lats, levels, hours, i_highest_level = read_raw_data(start_year, final_year)
+    check_for_missing_data(hours)
+
+
+
+    # Loop over all locations to write processed data to the output file.
+    counter = 0
+    total_iters = len(lats) * len(lons)
+    start_time = timer()
+
+    # Reading the data of all grid points from the NetCDF file all at once requires a lot of memory. On the other hand,
+    # reading the data of all grid points one by one takes up a lot of CPU. Therefore, the dataset is analysed in
+    # pieces: the subsets are read and processed consecutively.
+    n_subsets = int(np.ceil(float(len(lats)) / read_n_lats_at_once))
+
+    if subset_number_input == (-1):
+        i_subset = 0
+    elif subset_number_input >= n_subsets:
+        raise ValueError("User input subset number ({:.0f}) larger than total maximal subset index ({:.0f}) starting at 0.".format(subset_number_input, (n_subsets-1)))
+    else:
+        # User input given - processing only one latitude subset:
+        i_subset = subset_number_input
+        total_iters = len(lons)
```

Comment:
take this outside this function

Reply:
not sure why you ask that, I guess it might also be resolved by better function naming introduced in 68d38e1.